### PR TITLE
Port over charm lib changes from k8s charm relating to pod reschdule during cluster setup

### PR DESCRIPTION
**Pull request by shayancanonical**
Lib changes in https://github.com/canonical/mysql-k8s-operator/pull/499 to be published to charmhub

### CODE REVIEW

File: lib/charms/mysql/v0/mysql.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def wait_until_mysql_connection(self) -> None:
 # Increment this major API version when introducing breaking changes
 LIBAPI = 0
 
-LIBPATCH = 70
+LIBPATCH = 71
```
Comment (paulomach) [inline lib/charms/mysql/v0/mysql.py:137]: bump to `72` please

**Review by paulomach — COMMENTED** [review]
I've merged the admin_address, which will require some merge on the `mysql.py`. The merge should be smooth, but also it will require a change in how we build the connection string for mysqlsh calls

**Comment by taurus-forever** [comment]
Consider LGTM from me, but you need to address Paulo comment. I am not setting LGTM on GH to avoid the merge by mistake (2xLGTM already).

File: tests/unit/test_mysql.py [hunk 2]
```diff
@@ -463,6 +463,21 @@ def test_is_instance_configured_for_innodb(self, _run_mysqlsh_script):
         )
         self.assertFalse(is_instance_configured)
 
+    @patch("charms.mysql.v0.mysql.MySQLBase._run_mysqlsh_script")
+    def test_drop_group_replicatoin_metadata_schema(self, _run_mysqlsh_script):
```
Comment (paulomach) [inline tests/unit/test_mysql.py:467]: small typo

```suggestion
    def test_drop_group_replication_metadata_schema(self, _run_mysqlsh_script):
```

**Review by paulomach — APPROVED** [review]
nice!

File: tests/unit/test_mysql.py (hunk 2 above)
Comment (shayancanonical) [reply tests/unit/test_mysql.py]: addressed in 0718c94afe58b870cb573c0b59d72b3dd37212b8

**Pull request merged**